### PR TITLE
fix(npm): use shrinkwrap instead of lockfile

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

This renames our `package-lock.json` file to `npm-shrinkwrap.json`. `npm` treats the file almost identically during installations, but the only difference is that the shrinkwrap is actually published to the `npm` registry, which isn't recommended for JS libraries but is recommended for CLI tools[^1].

I only recently realized that `package-lock.json` isn't published to the `npm` registry, which explains why the dependency mismatch issue in https://github.com/readmeio/rdme/issues/703 happened. I suppose a downside of this is that if we have a security vulnerability in our dep tree, users are stuck with it?

## 🧬 QA & Testing

The shrinkwrap file [dates back to at least `npm@6`](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json) so it shouldn't change anything for CLI users that install/run `rdme` using Node 14+ (see how Node 14 tests continue to pass).

This is only an `npm` registry change so this has no affect on GitHub Actions users.

[The only users I'm worried about are `yarn` users](https://classic.yarnpkg.com/lang/en/docs/migrating-from-npm/) (we don't even tell people to install via `yarn`, but still). Once this change has been published to `npm` [on our `next` channel](https://www.npmjs.com/package/rdme?activeTab=versions), I'm going to try installing it via `yarn` and see what happens.

[^1]: You can read more about this in the `npm` docs: https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#package-lockjson-vs-npm-shrinkwrapjson

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ __tests__/__fixtures__/invalid-json/yikes.json
 CHANGELOG.md
 coverage/
 dist/
+npm-shrinkwrap.json
```
Comment (kanadgupta) [inline .prettierignore:5]: TIL that prettier ignores `package-lock.json` by default 🙄

File: npm-shrinkwrap.json
[the remark this answers is not in the document]
Comment (kanadgupta) [reply npm-shrinkwrap.json:1]: It updates the shrinkwrap[^1]! If both files (`package-lock` and `npm-shrinkwrap`) are present, the latter takes precedence and the former is ignored: https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#package-lockjson-vs-npm-shrinkwrapjson

[^1]: I confirmed this locally!

**Review by RyanGWU82 — APPROVED** [review]

**Comment by kanadgupta** [comment]
Confirmed that this plays nicely with `yarn` 🎊 

CleanShot 2023-04-19 at 15 50 06@2x

**Comment by domharrington** [comment]
This is weird, i thought npm basically deprecated the shrinkwrap file and preferred the package-lock ever since it came out 🤷‍♂️ but if it works, it works.

**Pull request merged**